### DISCUSSION TITLE
Template management for multi-site Wordpress

### DISCUSSION THREAD

**Discussion by verandesjardins** [elementor/elementor#24899] [Developers Community]
### Start a discussion

Hello,

I'm working on a plugin that will work on a multi-site Wordpress, and I have some questions : 

1. Could you please tell me what is the hook that is triggered when someone saves a "unique page template" ? To be more specific, I clicked on "Theme builder" and here, I want to add some templates for specific pages. And I want to be able to add some code, so that an action is performed when saving the template.
2. How can I export a page template programmatically ? For now, I go to "theme builder" and click on the 3 dots, and then click on "export". How can I do the same thing inside code ?
3. Last question : how can I "import"  this page template programmatically, into a subsite of my network sites ? I guess this generated json file can be put somewhere or imported by code ? 

Could you please help me ?

Thanks a lot.

**Comment by vladninja** [comment]
For 2 and 3 you can use a WP-CLI. We are using that for the same case as you (we are imported new templates for all of the websites in the multisite instance) https://developers.elementor.com/docs/cli/library-import/
p.s we are exporting the kit, made some changes in the templates and generate a kit and templates which we can import
